returns/transducers/tfilter.py

```diff
@@ -0,0 +1,41 @@
+from typing import Callable, TypeVar
+
+_ValueType = TypeVar('_ValueType')
+_AccValueType = TypeVar('_AccValueType')
+
+
+def tfilter(
+    predicate: Callable[[_ValueType], bool],
+) -> Callable[
+    [Callable[[_AccValueType, _ValueType], _AccValueType]],
+    Callable[[_AccValueType, _ValueType], _AccValueType],
+]:
+    """
+    :py:func:`filter <filter>` implementation on a transducer form.
+
+    .. code:: python
+
+      >>> from typing import List
+      >>> from returns.transducers import tfilter, reduce
```

Comment:
Let's import `reduce` from `functools`, because user might be confused: is it our `reduce` or default one?

Reply:
We need to use our `reduce` since we have more features than the built-in `reduce`